recipes/dbus/1.x.x/conanfile.py

```diff
@@ -70,52 +91,89 @@ def validate(self):
-        tc.variables["DBUS_SESSION_SOCKET_DIR"] = self.options.session_socket_dir
-        tc.generate()
-
-        deps = CMakeDeps(self)
-        deps.generate()
+        if self._meson_available:
+            tc = MesonToolchain(self)
+            tc.project_options["asserts"] = not is_apple_os(self)
+            tc.project_options["checks"] = False
+            tc.project_options["doxygen_docs"] = "disabled"
+            tc.project_options["modular_tests"] = "disabled"
+            tc.project_options["session_socket_dir"] = str(self.options.session_socket_dir)
+            tc.project_options["selinux"] = "enabled" if self.options.get_safe("with_selinux", False) else "disabled"
+            tc.project_options["systemd"] = "enabled" if self.options.get_safe("with_systemd", False) else "disabled"
+            if self.options.get_safe("with_systemd", False):
+                tc.project_options["systemd_system_unitdir"] = os.path.join(self.package_folder, "lib", "systemd", "system")
+                tc.project_options["systemd_user_unitdir"] = os.path.join(self.package_folder, "lib", "systemd", "user")
+            if is_apple_os(self):
+                tc.project_options["launchd_agent_dir"] = os.path.join(self.package_folder, "res", "LaunchAgents")
+            tc.project_options["x11_autolaunch"] = "enabled" if self.options.get_safe("with_x11", False) else "disabled"
+            tc.project_options["xml_docs"] = "disabled"
+            tc.generate()
+            env = VirtualBuildEnv(self)
+            env.generate(scope="build")
+        else:
+            tc = CMakeToolchain(self)
+            tc.cache_variables["DBUS_BUILD_TESTS"] = False
+            tc.cache_variables["DBUS_ENABLE_DOXYGEN_DOCS"] = False
+            tc.cache_variables["DBUS_ENABLE_XML_DOCS"] = False
+            tc.cache_variables["DBUS_BUILD_X11"] = bool(self.options.get_safe("with_x11", False))
+            tc.cache_variables["ENABLE_SYSTEMD"] = "ON" if self.options.get_safe("with_systemd", False) else "OFF"
+            tc.cache_variables["DBUS_WITH_GLIB"] = bool(self.options.get_safe("with_glib", False))
+            tc.cache_variables["DBUS_DISABLE_ASSERT"] = is_apple_os(self)
+            tc.cache_variables["DBUS_DISABLE_CHECKS"] = False
+
+            # Conan does not provide an EXPAT_LIBRARIES CMake variable for the Expat library.
+            # Define EXPAT_LIBRARIES to be the expat::expat target provided by Conan to fix linking.
+            tc.cache_variables["EXPAT_LIBRARIES"] = "expat::expat"
+
+            # https://github.com/freedesktop/dbus/commit/e827309976cab94c806fda20013915f1db2d4f5a
+            tc.cache_variables["DBUS_SESSION_SOCKET_DIR"] = str(self.options.session_socket_dir)
+            tc.generate()
+            cmake_deps = CMakeDeps(self)
+            cmake_deps.generate()
+        pkg_config_deps = PkgConfigDeps(self)
+        pkg_config_deps.generate()
```

Comment:
Is only meson using PkgConfigDeps? It could be moved to its context.

Reply:
Actually, the CMake build system requires it. I know they use it for at least detecting systemd support, if not more.

Reply:
Okay.